com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencePage.java

```diff
@@ -0,0 +1,112 @@
+package com.aobuchow.themes.spectrum.preferences;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.preference.PreferencePage;
+import org.eclipse.jface.resource.ColorRegistry;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.browser.IWebBrowser;
+
+public class SpectrumPreferencePage extends PreferencePage implements IWorkbenchPreferencePage {
+
+	private static final String rgbToHex = "#%02X%02X%02X";
+	private IWebBrowser browser;
+	private URL issuesURL;
+	private URL repoURL;
+	private ColorRegistry colorRegistry;
+	private Color accentColor;
+	private Color baseColor;
+	private Color backgroundColor;
+
+	@Override
+	public void init(IWorkbench workbench) {
+		colorRegistry = PlatformUI.getWorkbench().getThemeManager().getCurrentTheme().getColorRegistry();
```

Comment:
This will break if the user doesn’t have Spectrum Theme set as the current theme.